integration/deploy_manifest_test.go

```diff
@@ -0,0 +1,322 @@
+//go:build integration
+// +build integration
+
+// Copyright 2022 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/okteto/okteto/pkg/okteto"
+	"github.com/okteto/okteto/pkg/registry"
+)
+
+func TestDeployFromManifest(t *testing.T) {
+	ctx := context.Background()
+	oktetoPath, err := getOktetoPath(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	const (
+		gitRepo          = "git@github.com:okteto/go-getting-started.git"
+		repoDir          = "go-getting-started"
+		manifestFilename = "okteto.yml"
+		chartDir         = "chart"
+		chartFilename    = "Chart.yaml"
+		templateFilename = "k8s.yaml"
+		releaseName      = "hello-world"
+		manifestContent  = `build:
+  app:
+    context: .
+deploy:
+  - helm upgrade --install hello-world chart --set app.image=${build.app.image}`
+		chartContent = `apiVersion: v2
+name: hello-world
+description: A React application in Kubernetes
+type: application
+version: 0.1.0
+appVersion: 1.0.0`
+		templateContent = `apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: hello-world
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: hello-world
+  template:
+    metadata:
+      labels:
+        app: hello-world
+    spec:
+      containers:
+        - image: {{ .Values.app.image }}
+          name: hello-world
+
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: hello-world
+  annotations:
+    dev.okteto.com/auto-ingress: "true"
+spec:
+  type: ClusterIP
+  ports:
+    - name: "hello-world"
+      port: 8080
+  selector:
+    app: hello-world`
+	)
+
+	var (
+		testID          = strings.ToLower(fmt.Sprintf("DeployFromManifest-%s-%d", runtime.GOOS, time.Now().Unix()))
+		testNamespace   = fmt.Sprintf("%s-%s", testID, user)
+		expectedImage   = fmt.Sprintf("%s/%s/%s-app:okteto", okteto.Context().Registry, testNamespace, repoDir)
+		originNamespace = getCurrentNamespace()
+	)
+
+	if err := createNamespace(ctx, oktetoPath, testNamespace); err != nil {
```

Comment:
I'd move this function call after all the setup. So in case it fails (cloning the repo or writing the file) it doesn't create an empty namespace